docs/data/material/getting-started/templates/crud-dashboard/components/EmployeeEdit.js

```diff
@@ -0,0 +1,191 @@
+import PropTypes from 'prop-types';
+import Alert from '@mui/material/Alert';
+import Box from '@mui/material/Box';
+import CircularProgress from '@mui/material/CircularProgress';
+import { useNavigate, useParams } from 'react-router';
+import useNotifications from '../hooks/useNotifications/useNotifications';
+import {
+  getOne as getEmployee,
+  updateOne as updateEmployee,
+  validate as validateEmployee,
+} from '../data/employees';
+import EmployeeForm from './EmployeeForm';
+import PageContainer from './PageContainer';
+
+function EmployeeEditForm({ initialValues, onSubmit }) {
+  const { employeeId } = useParams();
+  const navigate = useNavigate();
+
+  const notifications = useNotifications();
+
+  const [formState, setFormState] = React.useState(() => ({
+    values: initialValues,
+    errors: {},
+  }));
+  const formValues = formState.values;
+  const formErrors = formState.errors;
+
+  const setFormValues = React.useCallback((newFormValues) => {
+    setFormState((previousState) => ({
+      ...previousState,
+      values: newFormValues,
+    }));
+  }, []);
+
+  const setFormErrors = React.useCallback((newFormErrors) => {
+    setFormState((previousState) => ({
+      ...previousState,
+      errors: newFormErrors,
+    }));
+  }, []);
+
+  const handleFormFieldChange = React.useCallback(
+    (name, value) => {
+      const validateField = async (values) => {
+        const { issues } = validateEmployee(values);
+        setFormErrors({
+          ...formErrors,
+          [name]: issues?.find((issue) => issue.path?.[0] === name)?.message,
+        });
+      };
+
+      const newFormValues = { ...formValues, [name]: value };
+
+      setFormValues(newFormValues);
+      validateField(newFormValues);
+    },
+    [formValues, formErrors, setFormErrors, setFormValues],
+  );
+
+  const handleFormReset = React.useCallback(() => {
+    setFormValues(initialValues);
+  }, [initialValues, setFormValues]);
+
+  const handleFormSubmit = React.useCallback(async () => {
+    const { issues } = validateEmployee(formValues);
+    if (issues && issues.length > 0) {
+      setFormErrors(
+        Object.fromEntries(issues.map((issue) => [issue.path?.[0], issue.message])),
+      );
+      return;
+    }
+    setFormErrors({});
+
+    try {
+      await onSubmit(formValues);
+      notifications.show('Employee edited successfully.', {
+        severity: 'success',
+        autoHideDuration: 3000,
+      });
+
+      navigate('/employees');
+    } catch (editError) {
+      notifications.show(`Failed to edit employee. Reason: ${editError.message}`, {
+        severity: 'error',
+        autoHideDuration: 3000,
+      });
+      throw editError;
+    }
+  }, [formValues, navigate, notifications, onSubmit, setFormErrors]);
+
+  return (
+    <EmployeeForm
+      formState={formState}
+      onFieldChange={handleFormFieldChange}
+      onSubmit={handleFormSubmit}
+      onReset={handleFormReset}
+      submitButtonLabel="Edit"
```

Comment:
"Edit" feels to me like it's going to bring me to a page to edit the record, but I'm already there

```suggestion
      submitButtonLabel="Save"
```